src/widget/wwidgetstack.cpp

```diff
@@ -85,13 +96,32 @@ void WWidgetStack::hideIndex(int index) {
     }
 }
 
+void WWidgetStack::showEvent(QShowEvent*) {
+    int index = static_cast<int>(m_currentPageControl.get());
+    setCurrentIndex(index);
+
+    // Set the page triggers to match the current index.
+    for (QMap<int, ControlObject*>::iterator it = m_triggers.begin();
+            it != m_triggers.end(); ++it) {
+        it.value()->set(it.key() == index ? 1.0 : 0.0);
```

Comment:
Mm I think there is missing a guard somehow. 
I have placed debug messages inside the showEvent  loop in showIndex and hideIndex and that prints the index and the function. 

I got this when fireing up Mixer: 
Debug [Main]: 1 WWidgetStack::showEvent 
Debug [Main]: 1 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showEvent 
Debug [Main]: 1 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showEvent 
Debug [Main]: 1 WWidgetStack::hideIndex 
Debug [Main]: 1 WWidgetStack::hideIndex 
Debug [Main]: 1 WWidgetStack::showEvent 
Debug [Main]: 1 WWidgetStack::hideIndex 
Debug [Main]: 1 WWidgetStack::hideIndex 
Debug [Main]: 0 WWidgetStack::showEvent 
Debug [Main]: 0 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showEvent 
Debug [Main]: 1 WWidgetStack::hideIndex 
Debug [Main]: 1 WWidgetStack::hideIndex 
Debug [Main]: 2 WWidgetStack::showEvent 
Debug [Main]: 2 WWidgetStack::showIndex 
Debug [Main]: 2 WWidgetStack::showIndex 
Debug [Main]: 3 WWidgetStack::showEvent 
Debug [Main]: 3 WWidgetStack::hideIndex 
Debug [Main]: 3 WWidgetStack::hideIndex 
Debug [Main]: Running Mixxx 

This is printed when enabling the Mixer

Debug [Main]: 3 WWidgetStack::showIndex 
Debug [Main]: 3 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showIndex 
Debug [Main]: 1 WWidgetStack::showEvent 
Debug [Main]: 1 WWidgetStack::hideIndex 
Debug [Main]: 1 WWidgetStack::hideIndex 
Debug [Main]: 2 WWidgetStack::showEvent 
Debug [Main]: 2 WWidgetStack::showIndex 
Debug [Main]: 2 WWidgetStack::showIndex 
Debug [Main]: 3 WWidgetStack::showEvent 
Debug [Main]: 3 WWidgetStack::hideIndex 
Debug [Main]: 3 WWidgetStack::hideIndex 
Debug [Main]: 1 WWidgetStack::showEvent 
Debug [Main]: 1 WWidgetStack::hideIndex 
Debug [Main]: 1 WWidgetStack::hideIndex 
Debug [Main]: 1 WWidgetStack::showEvent 
Debug [Main]: 1 WWidgetStack::hideIndex 
Debug [Main]: 1 WWidgetStack::hideIndex